### PR TITLE
Adding adcc

### PR DESCRIPTION
**Pull request by maxscheurer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by maxscheurer** [comment]
@conda-forge/help-python-c ready for review 🚀

**Comment by maxscheurer** [comment]
@conda-forge/staged-recipes ready for review 😄

File: recipes/adcc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    - adcc
+  commands:
+    - export PYLIB_EXT=`$PYTHON -c 'import sysconfig as s; print(s.get_config_var("EXT_SUFFIX"))'`
+    - test -f $PREFIX/lib/python{{ PY_VER }}/site-packages/libadcc$PYLIB_EXT
+
+about:
+    home: https://adc-connect.org
+    dev_url: https://github.com/adc-connect/adcc
+    doc_url: https://adc-connect.org
+    license: GPL-3.0-only
```
Comment (dopplershift) [inline recipes/adcc/meta.yaml:63]: Seems like "or later" is the [right one](https://github.com/adc-connect/adcc/blob/8b03f3cf057f588262bf24b5d0c843dcfed4f7a2/adcc/AdcMatrix.py#L11-L12):
```suggestion
    license: GPL-3.0-or-later
```

File: recipes/adcc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    license_family: GPL
```
Comment (dopplershift) [inline recipes/adcc/meta.yaml:64]:
```suggestion
    license_family: GPL3
```

File: recipes/adcc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    - pytest
+    - pytest-cov
+    - pyyaml
+    - pandas >=0.25.0
+    # Standard dependencies:
+    - numpy >=1.14
+    - h5py >=2.9
+    - scipy >=1.2
+    - tqdm >=4.30
+    - opt_einsum >=3.0
```
Comment (dopplershift) [inline recipes/adcc/meta.yaml:39]: I don't think any of these are used during package build.
```suggestion
```

File: recipes/adcc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+build:
+  number: 0
+  skip: true  # [win or py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - python
+    - libtensorlight >=3.0.1
```
Comment (dopplershift) [inline recipes/adcc/meta.yaml:20]: These are already covered by `host` below:
```suggestion
```

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/adcc/meta.yaml (hunk 3 above)
Comment (maxscheurer) [reply recipes/adcc/meta.yaml:39]: The test suite needs to be run in `build.sh`, that's why I put these dependencies in there.

File: recipes/adcc/meta.yaml (hunk 1 above)
Comment (maxscheurer) [reply recipes/adcc/meta.yaml:63]: Done.

File: recipes/adcc/meta.yaml (hunk 4 above)
Comment (maxscheurer) [reply recipes/adcc/meta.yaml]: Done.

**Comment by dopplershift** [comment]
Why run a test *before* building? That test could easily be done as part of the standard test phase.

**Comment by mfherbst** [comment]
We could have a restricted set of tests, that just run a calculation for a small basis set and checks the excitation energies (plain values) agree and maybe that the excitation vectors have a small residual. Than we don't have to download testdata at all, but still check a large part of the code is functioning properly. I would not patch that in, rather fix that directly upstream and just call pytest appropriately here.

How does that sound to everyone?

**Comment by maxscheurer** [comment]
I think this is a good suggestion. So then we include this in upstream adcc, release `v0.15.11` , and then continue here 😄

**Comment by maxscheurer** [comment]
@dopplershift We have adapted our test setup for c-f according to your suggestions (minimal tests can now be run during intended `test` phase). If you are happy with the current setup in general, we can release `v0.15.11` such that all pipelines here will work.

File: recipes/adcc/meta.yaml (hunk 3 above)
Comment (maxscheurer) [reply recipes/adcc/meta.yaml]: Done.

**Comment by dopplershift** [comment]
> The reason why it's not done in the usual test phase is that we don't install our package tests (they automatically download the reference data from our server, which we don't want to enable in every user's installation).

Just to clarify a point here. The standard `test` phase in the recipe is only run while running `conda build`, but *after* the package build. So it's not run on every users' system, only (generally) on conda-forge's CI systems.

In general, though, a reduced test set is great because you really don't need a full set here (that's what upstream's CI setup is for), just enough to validate that everything is built and linked correctly.

**Comment by maxscheurer** [comment]
> So it's not run on every users' system, only (generally) on conda-forge's CI systems.

Yes, I know. I meant we don't want to _install_ the entire test suite on users' machines, which would _enable_ them to (accidentally) over and over download testdata from our server. Sorry for the misunderstanding 😉
If you are comfortable with the current setup, we can deploy the most recent version (which has minimal tests) and fix pipelines here.

**Comment by maxscheurer** [comment]
@dopplershift, pipelines are working fine now, including minimal tests. So if we could get another review, that'd be great 👍

**Comment by maxscheurer** [comment]
@conda-forge/staged-recipes @dopplershift ready for review 🚀

**Comment by maxscheurer** [comment]
pinging @conda-forge/staged-recipes @dopplershift again, would be cool if this could be re-reviewed/merged at some point 😄 Thanks! 👍

**Pull request merged**